src/main/java/org/openbrewerydb/AppServer.java

```diff
@@ -35,6 +36,9 @@ public void run(final int port) {
         path(":id", () -> {
           get(this.breweryApi::getBrewery);
         });
+        path("search-nearest", () -> {
+          post(this.breweryApi::getNearestBreweries);
+        });
```

Comment:
I honestly don't have any experience with rest endpoints or the conventions here. But I want to be able to provide a request body so I assume it's ok to use a post?

Reply:
POST is what I've used in the past so I'm ok with it.

Reply:
I think this should be GET. 

https://blog.teamtreehouse.com/the-definitive-guide-to-get-vs-post

- Rule #1: Use GET for safe actions and POST for unsafe actions.
- Rule #2: Use POST when dealing with sensitive data.
- Rule #3: Use POST when dealing with long requests.

I don't think any of those apply, but let me know if I'm wrong.

Reply:
How do I pass in my object and radius fields as a request?

Reply:
Libraries don't allow passing in a request body for GET requests. Passing in as query parameters is I guess possible but not as user friendly IMO (in terms of input validation).

Reply:
@chrisjm this is a good read and is the same problem. Haha. 

https://dropbox.tech/developers/limitations-of-the-get-method-in-http

Reply:
I'm not really following the "not as user friendly" argument. The current API works with query parameters. Can you expand on that?

Reply:
Also, not sure I follow the relevance of the Dropbox article.

> The Dropbox API tries to use GET for calls that don't modify server state, but unfortunately this isn't always possible. GET requests don't have a request body, so all parameters must appear in the URL or in a header. While the HTTP standard doesn't define a limit for how long URLs or headers can be, mostHTTP clients and servers have a practical limit somewhere between 2 kB and 8 kB.

As far as I can tell we're only sending lat, long, and a number. What am I missing?

Reply:
I'm fine adding it as query params, but it does require the server to implement custom input validation logic whereas as a request body you effectively get your input validation for free via JSON deserialization.

Just provided the Dropbox link as an example from a top tech company where conventions can be broken.

And to reiterate, I don't use rest APIs so I myself am learning best practices.

Reply:
I think custom input validation is great. Rarely in the real world do you get anything for free :)

Reply:
I threw in a comment for the tests, but if the input fails validation, we should throw a 400.